### DISCUSSION TITLE
How to keep sensitive things like variables a secret?

### DISCUSSION THREAD

**Discussion by Rick-van-Dam** [qdm12/ddns-updater#673] [General]
I read the docs and iam wondering how to keep my cloudflare token a secret. I want to checkin the config file in git but obviously I don't want any secrets in there:

```
{
    "settings": [
      {
        "provider": "cloudflare",
        "zone_identifier": "some id",
        "domain": "domain.com",
        "host": "@",
        "ttl": 600,
        "email": "foo@bar.com",
        "key": "this should not be in git",
        "ip_version": "ipv4",
        "ipv6_suffix": ""
      }
    ]
  }
```

Is there any way to use something like environment tokens inside the settings json?

**Comment by qdm12** [comment]
Not at the moment, I would recommend just having the entire config file as a secret, especially since each provider has different credentials names etc. it would be rather cumbersome to have each as a unique key (through environment variables or other files).
